src/Kiota.Builder/Refiners/ShellRefiner.cs

```diff
@@ -102,26 +102,27 @@ private static void CreateCommandBuilders(CodeElement currentElement)
         {
             // Remove request executor
             RemoveUnusedParameters(currentClass);
-            // Replace Nav Properties with BuildXXXCommand methods
-            var navProperties = currentClass.GetChildElements().OfType<CodeProperty>().Where(e => e.IsOfKind(CodePropertyKind.RequestBuilder));
-            foreach (var navProp in navProperties)
-            {
-                var method = CreateBuildCommandMethod(navProp, currentClass);
-                currentClass.AddMethod(method);
-                currentClass.RemoveChildElement(navProp);
-            }
 
-            // Build command for indexers
-            var indexers = currentClass.GetChildElements().OfType<CodeIndexer>();
+            var children = currentClass.GetChildElements();
+            var indexers = children.OfType<CodeIndexer>();
             var classHasIndexers = indexers.Any();
-            CreateCommandBuildersFromIndexers(currentClass, indexers);
 
             // Clone executors & convert to build command
-            var requestExecutors = currentClass.GetChildElements().OfType<CodeMethod>().Where(e => e.IsOfKind(CodeMethodKind.RequestExecutor));
+            var requestExecutors = children.OfType<CodeMethod>()
+                .Where(e => e.IsOfKind(CodeMethodKind.RequestExecutor));
             CreateCommandBuildersFromRequestExecutors(currentClass, classHasIndexers, requestExecutors);
 
+            // Replace Nav Properties with BuildXXXCommand methods
+            var navProperties = children.OfType<CodeProperty>()
+                .Where(e => e.IsOfKind(CodePropertyKind.RequestBuilder));
```

Comment:
```suggestion
                .Where(static e => e.IsOfKind(CodePropertyKind.RequestBuilder));
```

Reply:
Is there a lint for this? I notice that there's a lot of reviews where you suggest it.

Reply:
I also notice the .Methods property performs ordering. Isn't that unnecessary computation that can be skipped here?

Reply:
> Is there a lint for this? I notice that there's a lot of reviews where you suggest it.

I've checked. There isn't any.

Reply:
thanks for checking and letting us know. Maybe a contribution to the CSharp analyzers for the next FHL? :)